feat(runtime): accept predicates in take and skip (#34657)
neovim/neovim: runtime/doc/lua.txt

        (`Iter`)

Iter:skip({n})                                                   *Iter:skip()*

>     Skips `n` values of an iterator pipeline, or the first values satisfying a

Reviewer: ```suggestion
    Skips `n` values of an iterator pipeline, or all values satisfying a
```
